sys/log/full/src/log_nmgr.c

```diff
@@ -99,6 +105,35 @@ log_nmgr_encode_entry(struct log *log, struct log_offset *log_offset,
         goto err;
     }
 
+#if MYNEWT_VAL(LOG_VERSION) > 2
+    switch (ueh.ue_etype) {
+    case LOG_ETYPE_STRING:
+        /* Trim string data to 128 characters to keep it consistent with v2 */
+        dlen = min(len-sizeof(ueh), 128);
+
+        rc = log_read(log, dptr, data, sizeof(ueh), dlen);
+        if (rc < 0) {
+            rc = OS_ENOENT;
+            goto err;
+        }
+        data[rc] = 0;
+        break;
+    case LOG_ETYPE_CBOR:
+        /* We can't trim CBOR stream - need to put it complete */
+        dlen = len - sizeof(ueh);
+        break;
+    case LOG_ETYPE_BINARY:
+        /* XXX just some placeholder for now, perhaps should be base64 encoded? */
+        strcpy(data, "<binary>");
+        dlen = strlen(data);
+        break;
+    default:
+        assert(0);
+        strcpy(data, "<??\?>");
```

Comment:
Accidental backslash? :)

Reply:
It's there to prevent `??>` being interpreted as trigraph. I could have used some other string though...

Reply:
Oh, right.  I totally missed that.